src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -7622,27 +7622,60 @@ void SketchObject::rebuildExternalGeometry()
                 BRepAdaptor_Curve curve(edge);
                 if (curve.GetType() == GeomAbs_Line) {
                     ExternalGeo.push_back(projectLine(curve, gPlane, invPlm));
+                    break;
+                }
+
+                gp_Dir vec1 = sketchPlane.Axis().Direction();
+                gp_Dir vec2;
+                if (curve.GetType() == GeomAbs_Circle) {
+                    vec2 = curve.Circle().Axis().Direction();
+                }
+                else if (curve.GetType() == GeomAbs_Ellipse) {
+                    vec2 = curve.Ellipse().Axis().Direction();
                 }
-                else if (curve.GetType() == GeomAbs_Circle) {
-                    gp_Dir vec1 = sketchPlane.Axis().Direction();
-                    gp_Dir vec2 = curve.Circle().Axis().Direction();
+
+                gp_Pnt beg, end;
+                if (curve.GetType() <= GeomAbs_Ellipse && !curve.IsClosed()) {

```

Comment:
`GeomAbs_Ellipse` is part of the opencascade api. The enum value for it changing would break so many things/files that it basically must remain stable. That said, I've switched to `(curve.GetType() == GeomAbs_Circle || curve.GetType() == GeomAbs_Ellipse)`.

Reply:
Unless that exact check is used within OCCT I would suggest to avoid it. Thanks for making the checks explicit.